backend/canisters/community/impl/src/updates/report_message.rs

```diff
@@ -1,21 +1,29 @@
-use crate::{read_state, run_regular_jobs, RuntimeState};
+use crate::activity_notifications::handle_activity_notification;
+use crate::{mutate_state, read_state, run_regular_jobs, RuntimeState};
 use canister_tracing_macros::trace;
 use chat_events::Reader;
 use community_canister::report_message::{Response::*, *};
 use ic_cdk_macros::update;
-use types::{CanisterId, Chat};
+use types::{CanisterId, Chat, UserId};
 use user_index_canister::c2c_report_message;
 
 #[update]
 #[trace]
 async fn report_message(args: Args) -> Response {
     run_regular_jobs();
 
-    let (c2c_args, user_index_canister) = match read_state(|state| build_c2c_args(args, state)) {
+    let (c2c_args, user_index_canister) = match read_state(|state| build_c2c_args(&args, state)) {
         Ok(ok) => ok,
         Err(response) => return response,
     };
 
+    if args.delete {
```

Comment:
Probably move this after the c2c call, otherwise if the c2c call fails the message will have been deleted but it can no longer be reported.